modules/charon-core/src/main/java/org/wso2/charon3/core/config/ExtensionBuilder.java

```diff
@@ -0,0 +1,495 @@
+                    }
+                } else {
+                    if (!(subAttribConfig.hasChildren())) {
+                        String error = "A attribute of complex type should have sub attributes";
+                        throw new InternalErrorException(error);
+                    } else {
+                        // need to build child schemas first
+                        buildComplexAttributeSchema(subAttribConfig, attributeSchemas, customConfig);
+                    }
+                }
+            }
+            // now all sub attributes must be already built
+            buildComplexSchema(config, attributeSchemas);
+        }
+    }
+
+    /*
+     * Has the logic to iterate through child attributes
+     *
+     * @param config
+     */
+    protected void buildComplexSchema(ExtensionAttributeSchemaConfig config,
+                                      Map<String, AttributeSchema> attributeSchemas) {
+
+        String[] subAttributeNames = config.getSubAttributes();
+        ArrayList<AttributeSchema> subAttributes = new ArrayList<AttributeSchema>();
+        for (String subAttributeName : subAttributeNames) {
+            subAttributes.add(attributeSchemas.get(getSubAttributeURI(config, subAttributeName)));
+        }
+        AttributeSchema complexAttribute = createSCIMAttributeSchema(config, subAttributes);
+        attributeSchemas.put(config.getURI(), complexAttribute);
+    }
+
+    /*
+     * Builds simple attribute schema
+     *
+     * @param config
+     */
+    protected void buildSimpleAttributeSchema(ExtensionAttributeSchemaConfig config,
+                                              Map<String, AttributeSchema> attributeSchemas) {
+
+        ArrayList<AttributeSchema> subAttributeList = new ArrayList<AttributeSchema>();
+        if (!attributeSchemas.containsKey(config.getURI())) {
+            AttributeSchema attributeSchema = createSCIMAttributeSchema(config, subAttributeList);
+            attributeSchemas.put(config.getURI(), attributeSchema);
+        }
+    }
+
+    /*
+     * create SCIM Attribute Schema
+     * @param attribute
+     * @param subAttributeList
+     * @return
+     */
+    public SCIMAttributeSchema createSCIMAttributeSchema(ExtensionAttributeSchemaConfig attribute,
+                                                         ArrayList<AttributeSchema> subAttributeList) {
+
+        return SCIMAttributeSchema.createSCIMAttributeSchema(attribute.getURI(), attribute.getName(),
+                attribute.getType(), attribute.getMultiValued(), attribute.getDescription(), attribute.getRequired(),
+                attribute.getCaseExact(), attribute.getMutability(), attribute.getReturned(),
+                attribute.getUniqueness(), attribute.canonicalValues, attribute.referenceTypes, subAttributeList);
+    }
+
+    /**
+     * Custom Attribute config class.
+     */
+    protected static class ExtensionAttributeSchemaConfig {
+
+        //unique identifier for the attribute
```

Comment:
Follow the comment pattern that we normally use. // Comment.

Reply:
fixed https://github.com/wso2/charon/pull/323/commits/6a773e3384a5db3baab94370f9f64a904834a48d